### PR TITLE
Update addon blueprint with ember-cli-update to 4.0.1

### PR DESCRIPTION
**Pull request by SergeAstapov**

### CODE REVIEW

**Comment by SergeAstapov** [comment]
Closes #338

File: .eslintrc.js [hunk 1]
```diff
@@ -25,23 +25,17 @@ module.exports = {
     // node files
     {
       files: [
-        '.eslintrc.js',
-        '.template-lintrc.js',
-        'ember-cli-build.js',
-        'index.js',
-        'setup-middleware.js',
-        'cli-options-filter.js',
-        'node-tests/setup-middleware-test.js',
```
Comment (scalvert) [inline .eslintrc.js:34]: Do we not need this anymore?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,80 @@
+name: CI
```
Comment (scalvert) [inline .github/workflows/ci.yml:1]: This name change will break our build badge in the README.md. Can we update that?

File: ember-cli-build.js [hunk 3]
```diff
@@ -18,12 +19,10 @@ module.exports = function (defaults) {
 
   const { maybeEmbroider } = require('@embroider/test-setup');
   return maybeEmbroider(app, {
-    packagerOptions: {
-      webpackConfig: {
-        node: {
-          crypto: 'empty',
```
Comment (scalvert) [inline ember-cli-build.js:24]: Is this not needed anymore?

File: package.json [hunk 4]
```diff
@@ -46,30 +46,33 @@
...
+    "test": "npm-run-all lint test:*",
     "test:node": "qunit node-tests",
     "test:ember": "ember test",
     "test:ember-compatibility": "ember try:each",
     "prepublishOnly": "ember ts:precompile",
     "postpublish": "ember ts:clean"
   },
   "resolutions": {
-    "socket.io": "^4.3.1"
+    "fuse.js": "6.4.6"
```
Comment (scalvert) [inline package.json:65]: Can you explain this?

File: package.json [hunk 5]
```diff
@@ -78,59 +81,58 @@
...
-    "eslint-config-prettier": "^8.0.0",
-    "eslint-plugin-ember": "^10.0.0",
+    "ember-try": "^2.0.0",
+    "eslint": "^7.32.0",
+    "eslint-config-prettier": "^8.3.0",
+    "eslint-plugin-ember": "^10.5.8",
     "eslint-plugin-node": "^11.1.0",
-    "eslint-plugin-prettier": "^3.1.4",
-    "express": "^4.17.1",
+    "eslint-plugin-prettier": "^4.0.0",
```
Comment (scalvert) [inline package.json:121]: I wonder if we should remove this in favor of using `prettier --check .`, which is [recommended](https://prettier.io/docs/en/integrating-with-linters.html#notes).

**Review by scalvert — CHANGES_REQUESTED** [review]

**Comment by scalvert** [comment]
@SergeAstapov thanks for doing this, and thanks for reaching out in discord!

File: .eslintrc.js (hunk 1 above)
Comment (SergeAstapov) [reply .eslintrc.js:34]: @scalvert it should be covered by more generic `'./node-tests/**/*.js',` few lines below

File: .github/workflows/ci.yml (hunk 2 above)
Comment (SergeAstapov) [reply .github/workflows/ci.yml:1]: @scalvert good point! Will update

File: ember-cli-build.js (hunk 3 above)
Comment (SergeAstapov) [reply ember-cli-build.js:24]: @scalvert actually this was giving me hard error:

```
Invalid configuration object. Webpack has been initialized using a configuration object that does not match the API schema.
51
 - configuration[0].node has an unknown property 'crypto'. These properties are valid:
52
   object { __dirname?, __filename?, global? }
53
   -> Options object for node compatibility features.
```

Looking into [PR where it was introduced](https://github.com/ember-a11y/ember-a11y-testing/pull/264) seem like it was needed to fix breaking build. but now it causes failure 🙃

File: package.json (hunk 4 above)
Comment (SergeAstapov) [reply package.json:65]: @scalvert got failure for the "Floating Dependencies" job in https://github.com/ember-a11y/ember-a11y-testing/runs/4608355862?check_suite_focus=true with error

```
error fuse.js@6.5.0: The engine "node" is incompatible with this module. Expected version ">=14". Got "12.22.1"
```

apparently fuse.js@6.5.0 had breaking change yesterday, there is logged issue https://github.com/krisk/Fuse/issues/600

Comment (SergeAstapov) [reply package.json:65]: previous `socket.io` resolutions seem not needed anymore

File: package.json (hunk 5 above)
Comment (SergeAstapov) [reply package.json:121]: @scalvert interesting! I think we may do so, this comes from default ember-cli blueprint, so not sure if it worth to diverge here

File: .github/workflows/ci.yml (hunk 2 above)
Comment (SergeAstapov) [reply .github/workflows/ci.yml:1]: @scalvert updated!

File: .eslintrc.js (hunk 1 above)
Comment (scalvert) [reply .eslintrc.js:34]: Ah sorry I missed that. Thanks!

File: .github/workflows/ci.yml (hunk 2 above)
Comment (scalvert) [reply .github/workflows/ci.yml:1]: Thank you!

File: ember-cli-build.js (hunk 3 above)
Comment (scalvert) [reply ember-cli-build.js:24]: Ya, we can remove it and see if it breaks anything :)

File: package.json (hunk 4 above)
Comment (scalvert) [reply package.json:65]: Makes sense, and thanks for the explanation.

File: package.json (hunk 5 above)
Comment (scalvert) [reply package.json:121]: Ya let's leave it for now, and we can work on updating the default blueprint.

**Review by scalvert — APPROVED** [review]

**Comment by scalvert** [comment]
Thanks again, @SergeAstapov!

File: package.json (hunk 4 above)
Comment (SergeAstapov) [reply package.json:65]: @scalvert fuse.js issue was resolved and new patch version released. Rised #352 to remove this resolution

**Pull request merged**